okta/resource_okta_user.go

```diff
@@ -291,7 +309,38 @@ func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
 		qp = query.NewQueryParams(query.WithActivate(false))
 	}
 
-	userBody := okta.User{Profile: profile}
+	password := d.Get("password").(string)
+	recoveryQuestion := d.Get("recovery_question").(string)
+	recoveryAnswer := d.Get("recovery_answer").(string)
+
+	if recoveryQuestion != "" && len(recoveryAnswer) < 4 {
+		return fmt.Errorf("[ERROR] Okta does not allow security answers with less than 4 characters")
+	}
+
+	p := &okta.PasswordCredential{
+		Value: password,
+	}
+
+	uc := &okta.UserCredentials{
+		Password: p,
+	}
+
+	if recoveryQuestion != "" && len(recoveryAnswer) >= 4 {
+		rq := &okta.RecoveryQuestionCredential{
+			Question: recoveryQuestion,
+			Answer:   recoveryAnswer,
+		}
+
+		uc = &okta.UserCredentials{
+			Password:         p,
+			RecoveryQuestion: rq,
+		}
+	}
+
+	userBody := okta.User{
+		Profile:     profile,
+		Credentials: uc,
+	}
```

Comment:
This can be cleaned up a bit, no need to instantiate the struct twice

```
	uc := &okta.UserCredentials{
		Password: &okta.PasswordCredential{
		        Value: password,
	        },
	}

	if recoveryQuestion != "" && len(recoveryAnswer) >= 4 {
		uc.RecoveryQuestion := &okta.RecoveryQuestionCredential{
			Question: recoveryQuestion,
			Answer:   recoveryAnswer,
		}
	}
```